main.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import os.path
+from argparse import Namespace
+
+ERR_MSG = "Error message: Invalid input. Make sure the input file is a valid csv file."
+
+
+def get_args() -> Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--input_file", type=str, required=True)
+    parser.add_argument("-o", "--output_file", type=str, required=True)
+    parser.add_argument("-r", "--row_limit", type=int, required=True)
+    return parser.parse_args()
+
+
+def is_input_valid(input_file: str, row_limit: int) -> bool:
+    is_valid_file = os.path.isfile(input_file) and os.path.exists(input_file)
+    is_csv = input_file.lower().endswith(".csv")
+    if not (is_valid_file and is_csv):
+        return False
+
+    with open(input_file, "r") as fp:
+        num_of_lines = len(fp.readlines())
+        return num_of_lines > row_limit
```

Comment:
What if it's equal to the row limit?
Also, I wouldn't say that this necessarily means the input was not valid.
I don't know if this was specifically mentioned in the assignment you were give, but a reasonable behavior in this case would be to read all of the lines in the file rather than only `row_limit` lines.

Reply:
I agree but this was indeed in the assignment description.